gobblin-core/src/main/java/gobblin/writer/objectstore/ObjectStoreDeleteOperation.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Copyright (C) 2014-2016 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+package gobblin.writer.objectstore;
+
+import java.io.IOException;
+
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+
+import com.google.common.base.Preconditions;
+import com.typesafe.config.Config;
+import com.typesafe.config.ConfigFactory;
+
+import gobblin.writer.objectstore.response.VoidResponse;
+
+/**
+ * An {@link ObjectStoreOperation} that deletes an object with <code>objectId</code> in the object store.
+ */
+@AllArgsConstructor(access=AccessLevel.PRIVATE)
+@Getter
+public class ObjectStoreDeleteOperation extends ObjectStoreOperation<VoidResponse> {
```

Comment:
I think we need DeleteResponse. For example, delete may succeed for fail depending if the store has the object or if there are not enough permissions. We can work on make DeleteResponse more precise but we need the extension point. For now, I suggest we have a return code if the operation succeeded or not.